src/c++/perf_analyzer/report_writer.cc

```diff
@@ -41,11 +41,14 @@ ReportWriter::Create(
     const std::vector<pa::PerfStatus>& summary, const bool verbose_csv,
     const bool include_server_stats, const int32_t percentile,
     const std::shared_ptr<ModelParser>& parser,
-    std::unique_ptr<ReportWriter>* writer, const bool should_output_metrics)
+    std::unique_ptr<ReportWriter>* writer, const bool should_output_metrics,
+    const std::shared_ptr<ProfileDataCollector>& collector,
```

Comment:
Why is this a reference? A shared pointer is already a pretty small structure to pass by copy. No need to optimize.

Reply:
Agreed. Surprised to find out there are a lot of discussions ([1](https://stackoverflow.com/questions/3310737/should-we-pass-a-shared-ptr-by-reference-or-by-value), [2](https://stackoverflow.com/questions/37610494/passing-const-shared-ptrt-versus-just-shared-ptrt-as-parameter)) around this topic on the internet, but all the argument for pass by reference is to avoid high overhead (of atomic de/increment), which is not our issue here.

Reply:
Yeah, usually we code for readability first, then we optimize if it's hurting user experience.